spring-data-jpa/src/test/java/org/springframework/data/jpa/repository/support/SimpleJpaRepositoryUnitTests.java

```diff
@@ -216,4 +223,20 @@ void applyQueryHintsToCountQueriesForSpecificationPageables() {
 
 		verify(metadata).getQueryHintsForCount();
 	}
+
+	@Test // GH-3188
+	void checkTransactionalAnnotation() {
+		Stream.of(SimpleJpaRepository.class.getDeclaredMethods()).filter(method -> Modifier.isPublic(method.getModifiers()) &&
```

Comment:
Care to rewrite the test in to a `@ParametrizedTest` with a `@MethodSource` supplying `Method`s as input argument to this test method?

Reply:
done.